packages/wonder-blocks-tooltip/components/tooltip-portal-mounter.test.js

```diff
@@ -14,119 +14,160 @@ import {TooltipPortalAttributeName} from "../util/constants.js";
 import TooltipPortalMounter from "./tooltip-portal-mounter.js";
 import TooltipPopper from "./tooltip-popper.js";
 
-import type {Placement} from "../util/types.js";
-
 type TestHarnessProps = {|
-    refChild: (Element | ?React.Component<*, *>) => mixed,
-    placement: Placement,
+    /**
+     * Identifier to give the bubble
+     */
+    bubbleId?: ?string,
+
+    /**
+     * Should we even have a bubble? When false, bubble is
+     * null.
+     * Defaults to true.
+     */
     hasBubble: boolean,
-|};
 
-type TestHarnessState = {
-    anchor: ?HTMLElement,
-};
+    /**
+     * Should we change the anchor element on each render?
+     * Defaults to false.
+     */
+    switchAnchor: boolean,
+
+    /**
+     * Should we even have an anchor? When false, anchor is
+     * null.
+     * Defaults to true.
+     */
+    hasAnchor: boolean,
+|};
 
 /**
- * Simple wrapper to wire up a TooltipPortal.
+ * Simple test harness to wire up the TooltipPortalMounter and provide
+ * some simple controls on behavior for use in testing.
  */
-class TestHarness extends React.Component<TestHarnessProps, TestHarnessState> {
+class TestHarness extends React.Component<TestHarnessProps> {
     static defaultProps = {
         hasBubble: true,
+        switchAnchor: false,
+        hasAnchor: true,
     };
 
-    constructor() {
-        super();
-    }
+    _lastAnchor: string = "anchorA";
+
+    _getAnchor() {
+        const {switchAnchor, hasAnchor} = this.props;
+        if (!hasAnchor) {
+            return ((null: any): React.Element<*>);
+        }
 
-    state = {anchor: null};
+        if (switchAnchor) {
+            this._lastAnchor =
+                this._lastAnchor === "anchorA" ? "anchorB" : "anchorA";
+        }
 
-    updateAnchorRef(node) {
-        if (node) {
-            const element = ReactDOM.findDOMNode(node);
-            if (this.state.anchor !== element) {
-                this.setState({anchor: ((element: any): ?HTMLElement)});
-            }
+        if (this._lastAnchor === "anchorA") {
+            return <View key={this._lastAnchor}>Anchor A</View>;
+        } else {
+            return <HeadingSmall key={this._lastAnchor}>Anchor B</HeadingSmall>;
```

Comment:
Interesting way of forcing prop updates.

Reply:
Yeah, may not be the best, but it was the first thing that came to mind.